src/Microsoft.TemplateEngine.Edge/Settings/TemplateInfo.cs

```diff
@@ -156,33 +84,53 @@ public string ShortName
 
         public IReadOnlyList<string> ShortNameList { get; set; }
 
-        [JsonProperty]
+        [JsonIgnore]
+        [Obsolete]
         public IReadOnlyDictionary<string, ICacheTag> Tags
         {
             get
             {
+                if (_tags == null)
```

Comment:
Nit: I like returning a known result immediately to keep the nesting depth low. Like:
```
if (_tags != null)
{
    return _tags;
}
.. rest of the logic.
```

But that is a personal preference. So, up to you.

Reply:
Applicable to other properties as well.